presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1095,7 +1097,12 @@ public void dropTable(HiveIdentity identity, String databaseName, String tableNa
                     .run("dropTable", stats.getDropTable().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient(identity)) {
                             Table table = client.getTable(databaseName, tableName);
-                            client.dropTable(databaseName, tableName, deleteData);
+                            EnvironmentContext environmentContext = new EnvironmentContext();
+                            if (purgeTableOnDrop) {
+                                environmentContext.putToProperties("ifPurge", "true");
+                            }
+
+                            client.dropTableWithEnvironmentContext(databaseName, tableName, deleteData, environmentContext);
```

Comment:
the line above (`.run("dropTable", stats.getDropTable()`) probably needs update

@electrum can you please confirm?